Kinvey.Tests/DataStoreTests/DataStoreNetworkIntegrationTests.cs

```diff
@@ -5089,204 +5234,247 @@ public async Task TestNetworkStoreGetSumAsync()
 
-			DataStore<ToDo> todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK, kinveyClient);
+            await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
 
-            await Assert.ThrowsExceptionAsync<KinveyException>(async delegate ()
-			{
-				await todoStore.PullAsync();
-			});
+            // Arrange
+            var newItem = new ToDo
+            {
+                Name = "Next Task",
+                Details = "A test",
+                DueDate = "2016-04-19T20:02:17.635Z"
+            };
+            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK);
 
-            await Assert.ThrowsExceptionAsync<KinveyException>(async delegate ()
-			{
-				await todoStore.PushAsync();
-			});
+            // Act
+            var savedToDo = await todoStore.SaveAsync(newItem);
 
-            await Assert.ThrowsExceptionAsync<KinveyException>(async delegate ()
-			{
-				await todoStore.SyncAsync();
-			});
-		}
+            var existingToDo = await todoStore.FindByIDAsync(savedToDo.ID);
+
+            // Teardown
+            await todoStore.RemoveAsync(savedToDo.ID);
+
+            // Assert
+            Assert.IsNotNull(existingToDo);
+            Assert.AreEqual(newItem.Name, existingToDo.Name);
+            Assert.AreEqual(newItem.Details, existingToDo.Details);
+            Assert.AreEqual(newItem.DueDate, existingToDo.DueDate);
+        }
 
         [TestMethod]
-        public async Task TestSubscribeUnsubscribeAsync()
+        public async Task TestSaveUpdateApiVersion4Async()
         {
             // Setup
+            kinveyClient = BuildClient("4");
+
             if (MockData)
             {
                 MockResponses(4);
             }
 
-            //Arrange
-            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK, kinveyClient);
-
             await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
 
-            await Client.SharedClient.ActiveUser.RegisterRealtimeAsync();
-
-            //Act
-            var isSuccess = await todoStore.Subscribe(new KinveyDataStoreDelegate<ToDo>
+            // Arrange
+            var newItem = new ToDo
             {
-                OnNext = (result) =>
-                {
-
-                },
-                OnStatus = (status) =>
-                {
-
-                },
-                OnError = (error) =>
-                {
+                ID = Guid.NewGuid().ToString(),
+                Name = "Next Task",
+                Details = "A test",
+                DueDate = "2016-04-19T20:02:17.635Z"
+            };
+            var todoStore = DataStore<ToDo>.Collection(collectionName, DataStoreType.NETWORK);
 
-                }
-            });
+            // Act
+            var savedToDo = await todoStore.SaveAsync(newItem);
 
-            await todoStore.Unsubscribe();
+            var existingToDo = await todoStore.FindByIDAsync(savedToDo.ID);
```

Comment:
I think it would be better to use just Find and assert the number of items - if for some reason two item is recorded multiple times

Reply:
Ok.